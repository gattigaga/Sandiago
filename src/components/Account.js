import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TouchableNativeFeedback } from "react-native";

const Container = styled.View`
  padding: 24px;
  border-bottom-width: 0.5;
  border-bottom-color: rgba(255, 255, 255, 0.2);
`;

const Name = styled.Text`
  font-family: Roboto-Bold;
  font-size: 18px;
  color: white;
  margin-bottom: 8px;
`;

const Description = styled.Text`
  font-family: Roboto-Regular;
  font-size: 14px;
  color: white;
`;

const Date = styled.Text`
  font-family: Roboto-Regular;
  font-size: 12px;
  color: rgba(255, 255, 255, 0.5);
  text-align: right;
  margin-bottom: 8px;
`;

const Account = ({ name, description, date, onPress }) => (
  <TouchableNativeFeedback onPress={onPress}>
    <Container>
      <Date>{date}</Date>
      <Name>{name}</Name>
      <Description>{description}</Description>
    </Container>
  </TouchableNativeFeedback>
);

Account.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.string,
  onPress: PropTypes.func
};

export default Account;
