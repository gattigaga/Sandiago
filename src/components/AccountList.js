import React from "react";
import { View } from "react-native";
import PropTypes from "prop-types";
import { format, isToday } from "date-fns";

import Account from "./Account";

const AccountList = ({ items, onPress }) => {
  return (
    <View>
      {items.map(item => {
        const schema = isToday(item.date)
          ? "hh:mm A"
          : "MMM DD, YYYY - hh:mm A";

        return (
          <Account
            key={item.id}
            name={item.name}
            description={item.description}
            date={format(item.date, schema)}
            onPress={() => onPress(item)}
          />
        );
      })}
    </View>
  );
};

AccountList.propTypes = {
  items: PropTypes.array,
  onPress: PropTypes.func
};

export default AccountList;
