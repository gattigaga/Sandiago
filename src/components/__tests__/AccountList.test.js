import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Account from "../Account";
import AccountList from "../AccountList";

describe("AccountList", () => {
  Date.now = jest.fn(() => 1529208317880);

  const setup = propOverrides => {
    const props = {
      items: [
        {
          id: 1,
          name: "Facebook",
          description: "This is my Facebook account.",
          date: Date.now()
        },
        {
          id: 2,
          name: "Twitter",
          description: "This is my Twitter account.",
          date: Date.now()
        },
        {
          id: 3,
          name: "Google",
          description: "This is my Google account.",
          date: Date.now()
        }
      ],
      onPress: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<AccountList {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onPress' while Account item pressed", () => {
    const { wrapper, props } = setup();

    wrapper
      .find(Account)
      .at(0)
      .simulate("press");

    expect(props.onPress).toBeCalledWith(props.items[0]);
  });
});
