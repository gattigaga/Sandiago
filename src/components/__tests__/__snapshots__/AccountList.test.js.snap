// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList should renders default 1`] = `
<Component>
  <Account
    date="11:05 AM"
    description="This is my Facebook account."
    key="1"
    name="Facebook"
    onPress={[Function]}
  />
  <Account
    date="11:05 AM"
    description="This is my Twitter account."
    key="2"
    name="Twitter"
    onPress={[Function]}
  />
  <Account
    date="11:05 AM"
    description="This is my Google account."
    key="3"
    name="Google"
    onPress={[Function]}
  />
</Component>
`;
